clients/src/main/java/org/apache/kafka/common/internals/ReflectiveStrategy.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.internals;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+/**
+ * A strategy which uses reflection to access methods without requiring them at compile-time.
+ */
+abstract class ReflectiveStrategy implements SecurityManagerCompatibility {
```

Comment:
This doesn't need to to be an abstract class. It doesn't have any fields. This can be achieve with static methods, no? And it avoids all of the complexity of inheritance and abstract classes.

Also this class doesn't really implement `SecurityManagerCompatibility`. It doesn't implement a single method of that interface.

Reply:
I was using this class to store common methods that were used by both the Legacy and Modern implementations, but not by Composite or Unsupported.

I converted it to a package-local util class instead.